modules/core/04-channel/v2/types/msgs.go

```diff
@@ -164,6 +168,10 @@ func (msg *MsgAcknowledgement) ValidateBasic() error {
 		return errorsmod.Wrap(commitmenttypesv1.ErrInvalidProof, "cannot submit an empty acknowledgement proof")
 	}
 
+	if len(msg.Packet.Data) != 1 {
+		return errorsmod.Wrapf(ErrInvalidPayload, "packet data must be of length 1, got %d instead", len(msg.Packet.Data))
+	}
```

Comment:
[q] why not just have this check in `Packet.ValidateBasic()`? we don't do other validation on the packet  directly here.

Just checked there actually, we already do this in the packet fn, the tests should still pass without these checks I think

Reply:
ah right I get you, yeah I guess I was thinking we would move the check to the packet.ValidateBasic(), and tests should pass once that is done. I think it's okay with the varargs as long as validate basic is != 1 and then eventually > 0.

Reply:
you can still always construct an invalid packet with the constructor, so I wouldn't be too concerned about allowing the construction of one with an empty payload list if it gets validated in ValidateBasic